pkg/resource/statefulset.go

```diff
@@ -157,6 +160,16 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 			TerminationGracePeriodSeconds: ptr.Int64(60),
 			Containers:                    b.MakeContainers(),
 			AutomountServiceAccountToken:  ptr.Bool(false),
+
+			// Setup the pod to not run as root
+			SecurityContext: &corev1.PodSecurityContext{
+				FSGroup:      ptr.Int64(1142),
```

Comment:
`FSGroup` may not run out of the box with openshift.  We need to make this configurable or remove.  It requires more testing as well.

Reply:
I moved this into a featuregate, so we can test it later